Tempus elaboration 29

Himanshu-it opened the pull request:
This PR contains:

1. Creation schema to persist model object definitions and related dao layer services.
2. Creation of CQL upgrade script for model object Definition.
3. Creation of  SQL upgrade script for model object Definition.

jetinder20 commented:
1. Add validations in service before saving/updating a model like if tenant exits etc (You can refer DeviceServiceImpl, Duplicate name etc for example) 

2. Don’t add the repeating tests in service tests which have already been tested in controllers(Saving model, retrieving) instead you can add corner cases like validations scenario in service as testing them through controller be over kill. 

3. You are not modifying the existing model, so you don’t need an upgrade scripts(add the queries in schema.sq;/cql directly) 

4. Please be consistent with names of services, objects etc. For example BaseModelObjectService could be ModelObjectServiceImpl. There are more places where naming can be better

5. I see potential problems with the model if they reflect what we need on UI.

a. ModelObject needs to have the DataModelId(on branch Tempus_17_and_29).
b. I dont think that we need customer id in that as customerId will have Datamodel already associated to it and model object needs to be associated with Datamodel (Point a covers that)
c.  We dont need "ModelObjectId" in the model explicitly as it can come from the interface.

Himanshu-it commented:
The recent commit contains, 
1. Adding validations and validation test cases.
2. Renaming modelObject to DataModelObject

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/hashmapinc/Tempus?pullRequest=424) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/hashmapinc/Tempus?pullRequest=424) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: Himanshu-it<br/>:x: anuj-akt<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/hashmapinc/Tempus?pullRequest=424) it.</sub>

jayeshkapadnis commented on common/data/src/main/java/com/hashmapinc/server/common/data/DataModelObject/AttributeDefinition.java:
```diff
@@ -0,0 +1,119 @@
+
+import java.util.UUID;
+
+public class AttributeDefinition {
```
Why isn't this inheriting BaseData ?

Himanshu-it commented:
In the recent commits:
1. Added a new field "type" in data_model_object table and removed tenant_Id.
2. Removed Id from attribute_definition table, instead used data_model_object_id and name as primary key.
